packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -423,7 +423,11 @@ export class AgentScheduler
 	}
 
 	private sendErrorEvent(eventName: string, error: any, key?: string) {
-		this.runtime.logger.sendErrorEvent({ eventName, key }, error);
+		const logger = createChildLogger({
+			logger: this.runtime.logger,
+			namespace: "fluid:AgentSchedulerDataStoreRuntime",
+		});
```

Comment:
Instead of creating a childLogger for every new event (this function gets called when we want to log one), we should create the childLogger once at the point where the AgentScheduler is created/initialized. In this case I think the constructor is a good place to do it, and here we can use `this.logger.sendErrorEvent(...)` or something similar.

Reply:
Updated